akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,12 +129,11 @@ object ActorMaterializer {
    * INTERNAL API
    */
   private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match {
+    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
```

Comment:
Yes, currently is saying "this might become public later". There is no full support for custom materializers yet, and this cast is just only one of the reasons.

But you say that this cast should be removed, which is basically excluding all Akka integration patterns, since our materializer will no longer to be able to access Akka provided features easily. The way to fix this is not to remove this method, but to eventually make ActorMaterializer properly supported and extensible. I.e. if you want to put a FIXME, then put it there instead.